### PR TITLE
[test] Multiple Sockets connections

### PR DESCRIPTION
**Pull request by viniciusfdasilva**
Implemented in the code unix domain, tcp, udp socket connections

### CODE REVIEW

File: .vscode/settings.json
Comment (ppenna) [inline .vscode/settings.json:1]: Normally you would add this file to .gitignore.

File: bin/server
Comment (ppenna) [inline bin/server:1]: Normally you would add this file to .gitignore.

File: client.c
Comment (ppenna) [inline client.c:1]: Normally you would add this file to .gitignore.

File: CMakeLists.txt
Comment (ppenna) [inline CMakeLists.txt:1]: You should add required compilation flags to enable all levels of optimizations. Also, compile with all warnings enabled.

File: src/server.c [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    
+}
+
+int main(int argc, char** argv)
+{
+
+    get_args(argc, argv);
+    attribuite_socket_type(socket_type);
+
+    int buffer[buffer_size];
```
Comment (ppenna) [inline src/server.c]: Allocate this on the heap instead.

File: src/server.c [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+{
+    perror(message);
+    exit(SYSTEM_EXIT_FAILED);
+}
+
+int create_socket()
+{
+    int server_socket;
+
+    server_socket = socket(sin_family, sock, IPPROTO_TCP);
```
Comment (ppenna) [inline src/server.c]: The third parameter is misleading and can be omitted.

File: src/client.c [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        struct sockaddr* address    = socket_type == UNIX_SOCKET_FLAG ? (struct sockaddr*)&server_address.socket_unix : (struct sockaddr*)&server_address.socket_ipv4;
+        socklen_t* server_addr_size = socket_type == UNIX_SOCKET_FLAG ? (socklen_t*)sizeof(server_address.socket_unix) : (socklen_t*)sizeof(server_address.socket_ipv4);
+        recvfrom(client_socket, received_buffer, buffer_size, 0, address, server_addr_size);
+    }
+
+    for(int i = 0; i < buffer_size; i++)
+    {
+        printf("%d", received_buffer[i]);
+    }
+    printf("\nRECEBIDO\n");
```
Comment (ppenna) [inline src/client.c]: Enclose this with a compiler guard. When running experiments, prints heavily impact performance.

File: src/client.c [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    
+}
+
+int main(int argc, char** argv)
+{
+
+    get_args(argc, argv);
+    attribuite_socket_type(socket_type);
+
+    int buffer[buffer_size];
```
Comment (ppenna) [inline src/client.c]: Allocate this on the heap.

File: src/client.c [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        buffer[i] = (int)(i % 255);
+    }
+    
+    int client_socket  = init_socket();
+    
+    server_address = config_server_address();
+
+    connect_to_server(client_socket);
+
+    socket_listen(client_socket, buffer, buffer_size);
```
Comment (ppenna) [inline src/client.c]: You are just sending/receiving one buffer. You should enclose this in a loop.

File: src/server.c [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+    int server_socket, client_socket;
+
+    server_socket = create_socket();
+
+    config_server_address();
+
+    bind_server(server_socket);
+    
+    server_listen(client_socket, server_socket, buffer_size);
```
Comment (ppenna) [inline src/server.c]: You are just sending/receiving one buffer. You should enclose this in a loop.

File: src/client.c (hunk 5 above)
Comment (viniciusfdasilva) [reply src/client.c]: I send multiples buffers on Python script that execute every socket tests, variating arguments and getting each execution time to generate final graphic.

File: src/server.c (hunk 6 above)
Comment (viniciusfdasilva) [reply src/server.c]: I send multiples buffers on Python script that execute every socket tests, variating arguments and getting each execution time to generate final graphic.

**Pull request merged**